test/CreatorToken.t.sol

```diff
@@ -49,6 +49,44 @@ contract CreatorTokenTest is Test {
     (uint256 _creatorFee, uint256 _adminFee) = creatorToken.calculateFees(_price);
     return _price + _creatorFee + _adminFee;
   }
+
+  function buyAToken(address _buyer) public {
+    (uint256 _tokenPrice, uint256 _creatorFee, uint256 _adminFee) = creatorToken.nextBuyPrice();
+    uint256 _totalPrice = _tokenPrice + _creatorFee + _adminFee;
+    uint256 originalCreatorTokenBalance = payToken.balanceOf(address(creatorToken));
+    uint256 originalBuyerBalance = creatorToken.balanceOf(_buyer);
+    uint256 originalTotalSupply = creatorToken.totalSupply();
+    deal(address(payToken), _buyer, _totalPrice);
+
+    vm.startPrank(_buyer);
+    payToken.approve(address(creatorToken), type(uint256).max);
+    creatorToken.buy(_totalPrice);
+    vm.stopPrank();
+
+    assertEq(creatorToken.balanceOf(_buyer), originalBuyerBalance + 1);
+    assertEq(creatorToken.ownerOf(creatorToken.lastId()), _buyer);
+    assertEq(payToken.balanceOf(address(creatorToken)), originalCreatorTokenBalance + _tokenPrice);
+    assertEq(creatorToken.totalSupply(), originalTotalSupply + 1);
+  }
+
+  function sellAToken(address _seller, uint256 _tokenId) public {
+    require(creatorToken.balanceOf(_seller) > 0, "seller does not own any tokens");
+    require(creatorToken.ownerOf(_tokenId) == _seller, "seller does not own the token");
```

Comment:
These two seem a bit redundant. Checking the seller owns the particular token ID should be sufficient, no?

Also, I'd consider adding something to the revert message to make it clear to the dev seeing the message that it's a broken *test* invariant. Maybe something like: `Broken test invariant: seller does not own the token to sell`.

Reply:
You're right! Updated the revert message here: https://github.com/showtime-xyz/creator-tokens/pull/14/commits/6a0ad6858e103fa6da6384492f59571f85783ba2